widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java

```diff
@@ -1476,6 +1509,9 @@ public void notificationsDialogDismissed() {
 
     public void queueForEngagementStarted() {
         Logger.d(TAG, "queueForEngagementStarted");
+        if (chatState.isOperatorOnline()) {
+            return;
+        }
```

Comment:
Could you please explain why we should not execute the queuing related code if an operator is online? Cannot get the point here.

Reply:
Because if the operator is online then that means that engagement has been already established and so no need to enqueue visitor for a new engagement.

Reply:
Unless I am missing something

Reply:
Hm, I don't know about this.
I would try to use `GliaEngagementStateRepository.hasOngoingEngagement()` function for this instead. However, if you are sure, then let's leave your approach.